UnityProject/Assets/Scripts/Tilemaps/Behaviours/Objects/RegisterPlayer.cs

```diff
@@ -291,13 +291,34 @@ public void ServerSlip(bool slipWhileWalking = false)
 			return;
 		}
 
-		ServerStun(StopMovement : false);
+		ServerSlip();
 		AudioSourceParameters audioSourceParameters = new AudioSourceParameters(pitch: Random.Range(0.9f, 1.1f));
 		SoundManager.PlayNetworkedAtPos(CommonSounds.Instance.Slip, WorldPositionServer, audioSourceParameters, sourceObj: gameObject);
 		// Let go of pulled items.
 		playerScript.objectPhysics.StopPulling(false);
 	}
 
+	private void ServerSlip()
+	{
+		ServerCheckStandingChange(true);
+		OnSlipChangeServer.Invoke(IsSlippingServer, true);
+		ServerDropItemsInHand();
+	}
+
+	[Server]
+	public void ServerDropItemsInHand()
+	{
+		foreach (var itemSlot in playerScript.DynamicItemStorage.GetNamedItemSlots(NamedSlot.leftHand))
+		{
+			Inventory.ServerDrop(itemSlot);
+		}
+
+		foreach (var itemSlot in playerScript.DynamicItemStorage.GetNamedItemSlots(NamedSlot.rightHand))
+		{
+			Inventory.ServerDrop(itemSlot);
+		}
+	}
```

Comment:
Would be useful to move to `DynamicItemStorage` if it doesn't already have one.

Reply:
Done